strawberryfields/api/connection.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+TODO
+"""
+from datetime import datetime
+import io
+import json
+import logging
+from typing import List
+
+import numpy as np
+import requests
+
+from strawberryfields.configuration import DEFAULT_CONFIG
+from strawberryfields.io import to_blackbird
+from strawberryfields.program import Program
+from .job import Job, JobStatus
+from .result import Result
+
+log = logging.getLogger(__name__)
+
+
+class RequestFailedError(Exception):
+    """Raised when a request to the remote platform returns an error response."""
+
+
+class Connection:
+    """Manages remote connections to the remote job execution platform and exposes
+    various job operations.
+
+    For basic usage, it is not necessary to manually instantiate this object; the user
+    is encouraged to use the higher-level interface provided by
+    :class:`~strawberryfields.engine.StarshipEngine`.
+
+    **Example:**
+
+    The following example instantiates a :class:`~strawberryfields.api.Connection` for a
+    given API authentication token, tests the connection, submits a new job, and
+    retrieves an existing job.
+
+    >>> connection = Connection(token="abc")
+    >>> success = connection.ping() # `True` if successful, `False` if the connection fails
+    >>> job = connection.create_job("chip2", program, shots=123)
+    >>> job
+    <Job: id=d177cbf5-1816-4779-802f-cef2c117dc1a, ...>
+    >>> job.status
+    <JobStatus.QUEUED: 'queued'>
+    >>> job.result
+    AttributeError
+    >>> job = connection.get_job(known_job_id)
+    >>> job
+    <Job: id=59a1c0b1-c6a7-4f9b-ae37-0ac5eec9c413, ...>
+    >>> job.status
+    <JobStatus.COMPLETED: 'complete'>
```

Comment:
see above comment

Reply:
This one is the actual string as implemented by `__str__`/`__repr__`.

Reply:
oh, for this I was referring to the comment about `<JobStatus.QUEUED: 'queued'>`